src/core/worker.c

```diff
@@ -271,7 +271,19 @@ QuicWorkerQueueConnection(
             "[conn][%p] Scheduling: %u",
             Connection,
             QUIC_SCHEDULE_QUEUED);
-        QuicConnAddRef(Connection, QUIC_CONN_REF_WORKER);
+        if (Connection->State.CleanupStarted) {
+#if DEBUG
+            //
+            // If the Connection is in the CleanupStarted state, its ref count
+            // is zero and it is queued here for freeing.
+            // Only increment the detailed ref count in debug builds.
+            // Just don't increment the main ref count in this state.
+            //
+            CxPlatRefIncrement(&Connection->RefTypeBiasedCount[QUIC_CONN_REF_WORKER]);
+#endif
+        } else {
+            QuicConnAddRef(Connection, QUIC_CONN_REF_WORKER);
+        }
```

Comment:
nit:
To make this easier to understand, I recommend starting with the golden path (here, what happen in non-debug scenarios). It also avoid an empty `if` body that some checks don't like.


```suggestion
        if (!Connection->State.CleanupStarted) {
            //
            // Only take a ref if the Connection is not in the CleanupStarted state.
            // It is only queued for freeing otherwise with a ref count of zero.
            //
            QuicConnAddRef(Connection, QUIC_CONN_REF_WORKER);
        }
#if DEBUG
        else {
            //
            // Only increment the detailed ref count.
            // 
            CxPlatRefIncrement(&Connection->RefTypeBiasedCount[QUIC_CONN_REF_WORKER]);
        }
#endif
```

And after writing that, I also think we should not increase the debug ref count because since we don't have a ref count. It would be confusing, the sum of the details would no longer match the refcount.

Reply:
ok, I'll rework things so that's not yet another thing to work around